.github/workflows/publish_docs.yml

```diff
@@ -1,57 +1,37 @@
 name: Publish docs to pages
 run-name: ${{ github.actor }} is publishing docs
 on:
-    # on a successful push
+    workflow_dispatch:  # Allow manual triggering
     push:
-        # only run on pushes to these branches
-        branches:
-            - 'master'
-            - 'main'
-            - 'docs'
-        # or run if pushes to these files.
-        # paths:
-        #    - 'docs/**.md'
-    check_run:
-        types:
-            # Only run if we have a successfully completed branch
-            - completed
-    # if desired, we can also use cron like rules to trigger these
-    # schedule:
-        #   - cron: 0 0 * * *
-    # Allow for manual triggering
-    workflow_dispatch:
+        branches: [master,main]
+        # paths: ['docs/**.md']  # API docs need building always
 
-# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
-permissions:
-    contents: read
-    pages: write
-    id-token: write
-
-# Allow only one concurrent deployment. Cancel any in-progress.
 concurrency:
-    group: "pages"
-    cancel-in-progress: true
+    group: github-pages
+    cancel-in-progress: false  # Skip any intermediate builds but finish deploying
 
 jobs:
     build:
         runs-on: ubuntu-latest
         env:
-            MDBOOK_ENV: 0.4.24
-            MERMAID_ENV: 0.12.6
+            MDBOOK_ENV: 0.4.40
+            MERMAID_ENV: 0.13.0
             DEST_DIR: /home/runner/.cargo/bin
         steps:
-            # these are other job descriptions to call.
-            - uses: actions/checkout@v3
-            - name: Configure rust  # TODO: can we export building rust and installing mdbook as an artifact and reuse it?
-              run: |
-                curl --proto '=https' --tlsv1.2 https://sh.rustup.rs -sSf -y | sh
-                rustup update
```

Comment:
huh, so `-y` is an abbreviation of `--speed-time` which (used to?) default to 1 second. Apparently, that's changed. (We want to be exceptionally strict about grabbing this script.)

We can probably replace it with the longer `--speed-time 1` which should cause a timeout if the server is unavailable or otherwise wonky.

I am somewhat curious, though. We should require `cargo docs` to build the source documentation, since we don't do that as part of the normal test/deploy cycle, and cargo was not installed on the base image. (or at least it wasn't when we originally built this.) I'll dig around and see what may have changed.

Reply:
If cargo doc fails, the workflow fails (=won't deploy broken / half empty / outdated), and will trigger failed CI pings so you'll know.

Reply:
I can't speak much about pre-focal times (only had xenial runners on some godfathered legacy stuff), but these are the rust toolchain versions pre-installed on the `@actions/runner-images:ubuntu` runners:

1. [Ubuntu2004-Readme#rust-tools](https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2004-Readme.md#rust-tools)
2. [Ubuntu2204-Readme#rust-tools](https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2204-Readme.md#rust-tools)
3. [Ubuntu2404-Readme#rust-tools](https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2404-Readme.md#rust-tools)

Reply:
Ah cool! Yeah the less "Let's run a random script downloaded from a cute, but potentially hostile domain service" the better.